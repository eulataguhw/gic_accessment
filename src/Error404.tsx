import React from "react";

export const Error404 = () => {
  return <div>404 Wrong URI</div>;
};
