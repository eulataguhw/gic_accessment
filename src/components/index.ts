export * from "./Buttons/TextButton";
export * from "./Containers/ContentContainer";
export * from "./Inputs/TextInput";
export * from "./Tables/CommonTable";
export * from "./Typographys/CustomTypography";
