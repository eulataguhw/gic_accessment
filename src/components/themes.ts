import { createTheme } from "@mui/material";

const theme = createTheme({
  cssVariables: true,
  palette: {
    primary: {
      main: "#3f51b5",
    },
    secondary: {
      main: "#f50057",
    },
  },
});

export default theme;
