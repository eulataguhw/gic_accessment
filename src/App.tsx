import "@fontsource/roboto/300.css";
import "@fontsource/roboto/400.css";
import "@fontsource/roboto/500.css";
import "@fontsource/roboto/700.css";
import "./App.css";
import React from "react";
import { BrowserRouter, Route, Routes } from "react-router";
import LandingScreen from "./views/LandingScreen/LandingScreen";
import Deposit from "./views/Deposit/Deposit";
import { Error404 } from "./Error404";
import BaseContainer from "./views/BaseContainer";
import Withdrawal from "./views/Withdrawal/Withdrawal";
import PrintStatement from "./views/PrintStatement/PrintStatement";
import Quit from "./views/Quit/Quit";
import { ThemeProvider } from "@emotion/react";
import { IntlProvider } from "react-intl";
import { Provider } from "react-redux";
import theme from "./components/themes";
import { store } from "./store/store";
import intl from "./shared/intl.json";

function App() {
  return (
    <Provider store={store}>
      <IntlProvider
        messages={intl}
        locale="en"
        defaultLocale="en"
      >
        <ThemeProvider theme={theme}>
          <BrowserRouter>
            <Routes>
              <Route
                path="/"
                element={<BaseContainer />}
              >
                <Route
                  path="/deposit"
                  element={<Deposit />}
                />
                <Route
                  path="/withdrawal"
                  element={<Withdrawal />}
                />
                <Route
                  path="/printStatement"
                  element={<PrintStatement />}
                />
                <Route
                  path="/landing"
                  element={<LandingScreen />}
                />
                <Route
                  path="/quit"
                  element={<Quit />}
                />
              </Route>

              <Route
                path="/*"
                element={<Error404 />}
              />
            </Routes>
          </BrowserRouter>
        </ThemeProvider>
      </IntlProvider>
    </Provider>
  );
}

export default App;

