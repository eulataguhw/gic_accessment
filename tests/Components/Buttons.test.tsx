import "@testing-library/jest-dom";

import { describe, expect, test } from "vitest";
import { render } from "@testing-library/react";
import React from "react";
import { TextButton } from "../../src/components";

describe("Testing Landing Screen", () => {
  test("initial render", () => {
    const { asFragment } = render(<TextButton>Initial Render</TextButton>);
    expect(asFragment).toMatchSnapshot();
  });
});
