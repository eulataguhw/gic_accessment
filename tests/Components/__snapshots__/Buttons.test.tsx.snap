// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing Landing Screen > initial render 1`] = `[Function]`;
